source/common/json/config_schemas.cc

```diff
@@ -1034,53 +1034,59 @@ const std::string Json::Schema::CLUSTER_SCHEMA(R"EOF(
           "binary" : {"type" : "string"}
         },
         "additionalProperties" : false
+      }
+    },
+    "type" : "object",
+    "properties" : {
+      "type" : {
+        "type" : "string",
+        "enum" : ["http", "redis", "tcp"]
       },
-      "health_check" : {
-        "type" : "object",
-        "properties" : {
-          "type" : {
-            "type" : "string",
-            "enum" : ["http", "tcp"]
-          },
-          "timeout_ms" : {
-            "type" : "integer",
-            "minimum" : 0,
-            "exclusiveMinimum" : true
-          },
-          "interval_ms" : {
-            "type" : "integer",
-            "minimum" : 0,
-            "exclusiveMinimum" : true
-          },
-          "unhealthy_threshold" : {
-            "type" : "integer",
-            "minimum" : 0,
-            "exclusiveMinimum" : true
-          },
-          "healthy_threshold" : {
-            "type" : "integer",
-            "minimum" : 0,
-            "exclusiveMinimum" : true
-          },
-          "path" : {"type" : "string"},
-          "send" : {
-            "type" : "array",
-            "items" : {"$ref" : "#/definitions/health_check_bytes"}
-          },
-          "receive" : {
-            "type" : "array",
-            "items" : {"$ref" : "#/definitions/health_check_bytes"}
-          },
-          "interval_jitter_ms" : {
-            "type" : "integer",
-            "minimum" : 0,
-            "exclusiveMinimum" : true
-          },
-          "service_name" : {"type" : "string"}
-        },
-        "required" : ["type", "timeout_ms", "interval_ms", "unhealthy_threshold", "healthy_threshold"],
-        "additionalProperties" : false
+      "timeout_ms" : {
+        "type" : "integer",
+        "minimum" : 0,
+        "exclusiveMinimum" : true
+      },
+      "interval_ms" : {
+        "type" : "integer",
+        "minimum" : 0,
+        "exclusiveMinimum" : true
+      },
+      "unhealthy_threshold" : {
+        "type" : "integer",
+        "minimum" : 0,
+        "exclusiveMinimum" : true
+      },
+      "healthy_threshold" : {
+        "type" : "integer",
+        "minimum" : 0,
+        "exclusiveMinimum" : true
+      },
+      "path" : {"type" : "string"},
+      "send" : {
+        "type" : "array",
+        "items" : {"$ref" : "#/definitions/health_check_bytes"}
       },
+      "receive" : {
+        "type" : "array",
+        "items" : {"$ref" : "#/definitions/health_check_bytes"}
+      },
```

Comment:
I would prefer to not make changers like this in this PR. I only added the TCP HC documentation because it was missing and was only in code. I will fix the docs for your other comments.